lib/jsonapi-serializers/dynamic_proxy_object.rb

```diff
@@ -0,0 +1,24 @@
+module JSONAPI
+	class DynamicProxyObject < BasicObject
+
+    def initialize(target)
+      @cache = {}
+      @target = target
+    end
+
+    def class
+      @target.class
+    end
+
+    def cache_key(name, args)
+      "#{name}_#{args.hash}"
+    end
+
+    def method_missing(name, *args, &block)
+      key = cache_key(name, args)
+      @cache[key] ||= @target.send(name, *args, &block)
+      @cache[key]
```

Comment:
This line can be deleted. All ruby `...=...` syntax always returns the right most expression.